File: backend/Dtos/FavoriteDto.cs
namespace backend.Dtos
{
        //Response — returned when listing favorites
        public class FavoriteResponseDto
        {
            public ItemDto Item { get; set; } = null!;
            public bool NotifyWhenAvailable { get; set; }
            public DateTime SavedAt { get; set; }
        }

        //Request — PATCH /api/favorites/{itemId}/notify
        public class ToggleNotifyDto
        {
            public bool NotifyWhenAvailable { get; set; }
        }
   
}

File: frontend/src/app/dtos/favoriteDTO.ts
import { ItemDTO } from "./itemDTO";

export namespace FavoriteDTO {
  export interface FavoriteResponseDTO {
    item: ItemDTO.ItemSummaryDTO;
    notifyWhenAvailable: boolean;
    savedAt: string;
  }
 
  export interface ToggleNotifyDTO {
    notifyWhenAvailable: boolean;
  }
}